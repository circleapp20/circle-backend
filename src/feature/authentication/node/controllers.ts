import { Response } from 'express';
import {
	checkUserVerificationCode,
	createUserProfileWithDefaultValues,
	getUserAccountWithCredentials,
	getUserProfileById,
	verifyUserLoginCredentials
} from 'feature/authentication/node/dataService';
import { sendVerificationCodeByMedia } from 'feature/authentication/node/messaging';
import { getResponseData } from 'shared/common/utilities';
import { Constants } from 'shared/constants';
import { IAuthUser, IRequest } from 'shared/types';

export const verifyUserCredentials = async (req: IRequest, res: Response) => {
	const user = await createUserProfileWithDefaultValues(req.body.data);

	sendVerificationCodeByMedia({
		media: user.email ? 'email' : 'phoneNumber',
		verificationCode: user.verificationCode,
		email: user.email,
		phoneNumber: user.phoneNumber
	});

	const responseData = getResponseData(user);
	res.status(Constants.status.CREATED).json(responseData);
};

export const verifyUserVerificationCode = async (req: IRequest, res: Response) => {
	const { id }: IAuthUser = req.user;
	const data = Object.assign({}, req.body.data, { id });
	const status = await checkUserVerificationCode(data);
	const responseData = getResponseData(status);
	res.status(Constants.status.CREATED).json(responseData);
};

export const verifyUserLogin = async (req: IRequest, res: Response) => {
	const user = await verifyUserLoginCredentials(req.body.data);
	const responseData = getResponseData(user);
	res.status(Constants.status.CREATED).json(responseData);
};

export const resendUserVerificationCode = async (req: IRequest, res: Response) => {
	const { media }: any = req.query;
	const user = await getUserProfileById(req.user.id);

	sendVerificationCodeByMedia({
		media: media ? media : user.email ? 'email' : 'phoneNumber',
		verificationCode: user.verificationCode,
		email: user.email,
		phoneNumber: user.phoneNumber
	});

	const responseData = getResponseData(true);
	res.status(Constants.status.CREATED).json(responseData);
};

/**
 * the controller verifies the user's credentials in the database
 * and sends the verification code to the user by email or password
 * should the user have both email and password, a response of message
 * is return to the user to request which media to send the verification
 * code to either email or phone number
 */
export const verifyUserCredentialsForPasswordReset = async (req: IRequest, res: Response) => {
	const profile = await getUserAccountWithCredentials(req.body.data);

	const { email, verificationCode, phoneNumber } = profile;
	let message = 'Verification code cannot be sent. User has both email and phone number';

	sendVerificationCodeByMedia({
		media: phoneNumber ? 'phoneNumber' : 'email',
		verificationCode,
		email: email && !phoneNumber ? email : '',
		phoneNumber: !email && phoneNumber ? phoneNumber : ''
	});

	if (email && !phoneNumber) {
		message = "Verification code sent to user's email";
	} else if (!email && phoneNumber) {
		message = "Verification code sent to user's phone number";
	}

	const responseData = getResponseData({ user: profile, message });
	res.status(Constants.status.CREATED).json(responseData);
};
