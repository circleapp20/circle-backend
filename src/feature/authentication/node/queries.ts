import { Users } from 'shared/common/schema/users';
import { EntityManager } from 'typeorm';

export const countMatchingIdAndCodeQuery = (
	manager: EntityManager,
	id: string,
	verificationCode: string
) => {
	return manager
		.createQueryBuilder(Users, 'u')
		.where('u.id = :id', { id })
		.andWhere('u.verificationCode = :code', { code: verificationCode })
		.getCount();
};

export const updateUserVerificationCodeQuery = (
	manager: EntityManager,
	values: { id: string; verificationCode: string }
) => {
	const { id, verificationCode } = values;
	return manager
		.createQueryBuilder(Users, 'u')
		.update()
		.set({ verificationCode })
		.where('u.id = :id', { id })
		.execute();
};
