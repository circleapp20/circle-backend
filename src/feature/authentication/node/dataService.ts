import bcryptjs from 'bcryptjs';
import {
	countMatchingIdAndCodeQuery,
	updateUserVerificationCodeQuery
} from 'feature/authentication/node/queries';
import { getBadRequestError } from 'shared/common/errors';
import { generateCodeFromNumber } from 'shared/common/utilities';
import { Constants } from 'shared/constants';
import { runInsertQuery, runInTransaction, runQuery } from 'shared/node/database';
import {
	addUserProfileQuery,
	getUserByCredentialsQuery,
	getUserByIdQuery
} from 'shared/node/queries';
import { getSignedAuthToken } from 'shared/node/validation';
import { IAddUserProfile } from 'shared/types';
import { EntityManager } from 'typeorm';

export const addUserTransaction = (email = '', phoneNumber = '') => {
	return async (manager: EntityManager) => {
		// generate a verification code for the user
		const verificationCode = generateCodeFromNumber();

		// construct a default user info
		const profile: IAddUserProfile = {
			username: '',
			password: '',
			dob: new Date(),
			image: '',
			biography: '',
			email: email || '',
			phoneNumber: phoneNumber || '',
			isEmailVerified: false,
			verificationCode,
			roles: [Constants.privileges.USER],
			name: ''
		};

		const [user] = await runInsertQuery(addUserProfileQuery, [profile], manager);

		return runQuery(getUserByIdQuery, [user.id], manager);
	};
};

export const createUserProfileWithDefaultValues = async (data: {
	email?: string;
	phoneNumber?: string;
}) => {
	const user = await runQuery(getUserByCredentialsQuery, [data]);
	if (user) throw getBadRequestError('User already exists');

	// save user profile into the database
	const userProfile = await runInTransaction(addUserTransaction(data.email, data.phoneNumber));

	// add token to user profile
	const token = getSignedAuthToken({ id: userProfile!.id, roles: [Constants.privileges.USER] });
	const profile = Object.assign({}, userProfile, { token });

	return profile;
};

export const checkUserVerificationCode = async (data: { id: string; verificationCode: string }) => {
	const { id, verificationCode } = data;

	if (!id || !verificationCode) {
		throw getBadRequestError('user id and verification code are required');
	}

	const count = await runQuery(countMatchingIdAndCodeQuery, [id, verificationCode]);
	if (!count) throw getBadRequestError('invalid verification code');

	return true;
};

export const verifyUserLoginCredentials = async (data: {
	email?: string;
	phoneNumber?: string;
	username?: string;
	password: string;
}) => {
	const { phoneNumber, password, email, username } = data;
	if (!phoneNumber && !email && !username) {
		throw getBadRequestError('phoneNumber, email or username is required');
	}
	if (!password) throw getBadRequestError('password is required');

	const user = await runQuery(getUserByCredentialsQuery, [data]);
	if (!user) throw getBadRequestError('Invalid user account');

	if (!bcryptjs.compareSync(password, user.password)) {
		throw getBadRequestError('Invalid user account');
	}

	const { password: hashedPassword, ...other } = user;

	// create user token
	const token = getSignedAuthToken({ id: user.id, roles: user.roles });
	const profile = Object.assign({}, other, { token });

	return profile;
};

export const getUserProfileById = async (id: string) => {
	const user = await runQuery(getUserByIdQuery, [id]);
	if (!user) throw getBadRequestError('Invalid user account');
	return user;
};

/**
 * the service function validates the user's credentials in the
 * database to check if the user exists
 * if the user does exists, a new verification code is generated
 * and stored in the database
 */
export const getUserAccountWithCredentials = async (args: {
	email?: string;
	phoneNumber?: string;
	username?: string;
}) => {
	const user = await runQuery(getUserByCredentialsQuery, [args]);
	if (!user) throw getBadRequestError('Invalid user account');

	// generate new verification code
	const verificationCode = generateCodeFromNumber();

	// update user profile
	await runQuery(updateUserVerificationCodeQuery, [{ id: user.id, verificationCode }]);

	const { password, ...other } = user;

	const token = getSignedAuthToken({ id: user.id, roles: user.roles });
	const profile = Object.assign({}, other, { token, verificationCode });

	return profile;
};
