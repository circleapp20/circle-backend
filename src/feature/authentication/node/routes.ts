import * as schema from 'feature/authentication/common/dataSchema';
import * as controller from 'feature/authentication/node/controllers';
import { Constants } from 'shared/constants';
import { IApiRoute } from 'shared/types';

export const authRoutes: IApiRoute[] = [
	{
		path: '/auth/sign-up',
		method: 'post',
		controller: controller.verifyUserCredentials,
		schema: schema.VerifyUserCredentialsSchema
	},
	{
		path: '/auth/code',
		method: 'post',
		controller: controller.verifyUserVerificationCode,
		privileges: [Constants.privileges.SUPER_ADMIN, Constants.privileges.USER],
		schema: schema.VerifyVerificationCodeSchema
	},
	{
		path: '/auth/sign-in',
		method: 'post',
		controller: controller.verifyUserLogin,
		schema: schema.VerifyUserLoginSchema
	},
	{
		path: '/auth/code/resend',
		method: 'post',
		controller: controller.resendUserVerificationCode,
		privileges: [Constants.privileges.USER]
	},
	{
		path: '/auth/credentials',
		method: 'post',
		controller: controller.verifyUserCredentialsForPasswordReset
	}
];
