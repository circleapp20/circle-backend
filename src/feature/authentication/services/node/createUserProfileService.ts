import { getBadRequestError } from 'base/common/errors';
import { generateCodeFromNumber } from 'base/common/utilities';
import { Constants } from 'base/constants';
import * as sharedQueries from 'base/node/queries';
import { getSignedAuthToken } from 'base/node/validation';
import { IAddUserProfile } from 'base/types';
import { runInsertQuery, runInTransaction, runQuery } from 'core/node/database/queryRunners';
import { decryptData, encryptData } from 'core/node/encryption';
import { EntityManager } from 'typeorm';

export const addUserTransaction = (email = '', phoneNumber = '') => {
	return async (manager: EntityManager) => {
		// generate a verification code for the user
		const verificationCode = generateCodeFromNumber();
		const encryptedVerificationCode = encryptData({ text: verificationCode });

		// construct a default user info
		const profile: IAddUserProfile = {
			username: '',
			password: '',
			dob: new Date(),
			image: '',
			biography: '',
			email: email || '',
			phoneNumber: phoneNumber || '',
			isEmailVerified: false,
			verificationCode: encryptedVerificationCode,
			roles: [Constants.privileges.USER],
			name: ''
		};

		const [user] = await runInsertQuery(sharedQueries.addUserProfileQuery, [profile], manager);

		return runQuery(sharedQueries.getUserByIdQuery, [user.id], manager);
	};
};

export const createUserProfileWithDefaultValues = async (data: {
	email?: string;
	phoneNumber?: string;
}) => {
	const user = await runQuery(sharedQueries.getUserByCredentialsQuery, [data]);
	if (user) throw getBadRequestError('User already exists');

	// save user profile into the database
	const userProfile = await runInTransaction(addUserTransaction(data.email, data.phoneNumber));
	const verificationCode = decryptData({ encryptedText: userProfile!.verificationCode });

	// add token to user profile
	const token = getSignedAuthToken({ id: userProfile!.id, roles: [Constants.privileges.USER] });
	const profile = Object.assign({}, userProfile, { token, verificationCode });

	return profile;
};
