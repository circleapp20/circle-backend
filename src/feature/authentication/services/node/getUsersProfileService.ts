import { getBadRequestError } from 'base/errors/node/badRequestError';
import { runQuery } from 'core/database/queryRunners';
import { decryptData } from 'core/encryption/node/encryption';
import { getUserByIdQuery } from 'core/queries/userQueries';

export const getUserProfileById = async (id: string) => {
	const user = await runQuery(getUserByIdQuery, [id]);
	if (!user) throw getBadRequestError('Invalid user account');
	const { password, verificationCode, ...other } = user;
	const decryptedVerificationCode = decryptData({ encryptedText: verificationCode });
	return Object.assign({}, other, { verificationCode: decryptedVerificationCode });
};
