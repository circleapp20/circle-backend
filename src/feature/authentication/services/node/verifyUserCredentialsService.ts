import { getBadRequestError } from 'base/errors/node/badRequestError';
import { getSignedAuthToken } from 'base/server/validation';
import { generateCodeFromNumber } from 'base/utils/node/codeGenerator';
import bcryptjs from 'bcryptjs';
import { runQuery } from 'core/database/queryRunners';
import { decryptData, encryptData } from 'core/encryption/node/encryption';
import { getUserByCredentialsQuery, getUserByIdQuery } from 'core/queries/userQueries';
import { updateUserVerificationCodeQuery } from 'feature/authentication/queries/updateUserVerificationCodeQuery';

export const verifyUserLoginCredentials = async (data: {
	email?: string;
	phoneNumber?: string;
	username?: string;
	password: string;
}) => {
	const { phoneNumber, password, email, username } = data;
	if (!phoneNumber && !email && !username) {
		throw getBadRequestError('phoneNumber, email or username is required');
	}
	if (!password) throw getBadRequestError('password is required');

	const user = await runQuery(getUserByCredentialsQuery, [data]);
	if (!user) throw getBadRequestError('Invalid user account');

	if (!bcryptjs.compareSync(password, user.password)) {
		throw getBadRequestError('Invalid user account');
	}

	const { password: hashedPassword, verificationCode, ...other } = user;

	// create user token
	const token = getSignedAuthToken({ id: user.id, roles: user.roles });
	const profile = Object.assign({}, other, { token });

	return profile;
};

/**
 * the service function validates the user's credentials in the
 * database to check if the user exists
 * if the user does exists, a new verification code is generated
 * and stored in the database
 */
export const getUserAccountWithCredentials = async (args: {
	email?: string;
	phoneNumber?: string;
	username?: string;
}) => {
	const user = await runQuery(getUserByCredentialsQuery, [args]);
	if (!user) throw getBadRequestError('Invalid user account');

	// generate new verification code
	const verificationCode = generateCodeFromNumber();
	const encryptedVerificationCode = encryptData({
		text: verificationCode
	});

	// update user profile
	await runQuery(updateUserVerificationCodeQuery, [
		{ id: user.id, verificationCode: encryptedVerificationCode }
	]);

	const { password, ...other } = user;

	const token = getSignedAuthToken({ id: user.id, roles: user.roles });
	const profile = Object.assign({}, other, { token, verificationCode });

	return profile;
};

export const checkUserVerificationCode = async (data: { id: string; verificationCode: string }) => {
	const { id, verificationCode } = data;

	if (!id || !verificationCode) {
		throw getBadRequestError('user id and verification code are required');
	}

	const user = await runQuery(getUserByIdQuery, [id]);
	if (!user) throw getBadRequestError('Invalid user account');

	const decryptedVerificationCode = decryptData({ encryptedText: user.verificationCode });
	if (verificationCode !== decryptedVerificationCode) {
		throw getBadRequestError('Invalid verification code');
	}

	return true;
};
