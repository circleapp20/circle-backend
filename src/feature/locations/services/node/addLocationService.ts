import { Constants } from 'base/constants';
import { runInsertQuery, runInTransaction, runQuery } from 'core/node/database/queryRunners';
import {
	addLocationPlaceQuery,
	addLocationQuery
} from 'feature/locations/queries/addLocationQuery';
import { getLocationByIdQuery } from 'feature/locations/queries/getLocationQueries';
import { EntityManager } from 'typeorm';

interface IAddLocation {
	name: string;
	address: string;
	latitude: number;
	longitude: number;
	isVerified?: boolean;
	placeId: string;
}

export const addLocationTransaction = (args: IAddLocation) => {
	const { placeId, ...other } = args;
	return async (manager: EntityManager) => {
		const [location] = await runInsertQuery(addLocationQuery, [other], manager);
		if (placeId) await runQuery(addLocationPlaceQuery, [{ id: location.id, placeId }], manager);
		return runQuery(getLocationByIdQuery, [location.id], manager);
	};
};

export const addLocationService = async (args: IAddLocation, privileges: string[]) => {
	const isVerified = privileges.includes(Constants.privileges.LEAD_FELLOW);
	const locationTransaction = addLocationTransaction({ ...args, isVerified });
	const location = await runInTransaction(locationTransaction);
	return location;
};
