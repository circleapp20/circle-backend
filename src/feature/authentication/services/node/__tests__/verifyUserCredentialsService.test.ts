import { entityManager } from 'base/testUtils/node/entityManager';
import * as utils from 'base/utils/node/codeGenerator';
import bcryptjs from 'bcryptjs';
import { runQuery } from 'core/database/queryRunners';
import * as encryption from 'core/encryption/node/encryption';
import {
	checkUserVerificationCode,
	getUserAccountWithCredentials,
	verifyUserLoginCredentials
} from 'feature/authentication/services/node/verifyUserCredentialsService';
import { createUserFixture } from 'fixtures/users';

jest.mock('core/database/queryRunners');
jest.mock('core/models/node/users');
jest.mock('core/models/node/fellows');

beforeEach(() => jest.clearAllMocks());

describe('#checkUserVerificationCode', () => {
	test('should return true if code and id matches', async () => {
		const user = createUserFixture();
		entityManager.getOne.mockReturnValueOnce({
			...user,
			verificationCode: encryption.encryptData({ text: '345287' })
		});
		const data = { id: user.id, verificationCode: '345287' };
		const results = await checkUserVerificationCode(data);
		expect(results).toBeTruthy();
	});

	test('should throw if id is empty', () => {
		const data = { id: '', verificationCode: '3452' };
		expect(checkUserVerificationCode(data)).rejects.toThrowErrorMatchingSnapshot();
	});

	test('should throw if verificationCode is empty', () => {
		const data = { id: 'x7i9-3l-n3k4-3i8bi2', verificationCode: '' };
		expect(checkUserVerificationCode(data)).rejects.toThrowErrorMatchingSnapshot();
	});

	test('should throw if user is null', () => {
		entityManager.getOne.mockReturnValueOnce(null);
		const data = { id: 'x7i9-3l-n3k4-3i8bi2', verificationCode: '345287' };
		expect(checkUserVerificationCode(data)).rejects.toThrowErrorMatchingSnapshot();
	});

	test('should decrypt stored encrypted verification code', async () => {
		const user = createUserFixture();
		entityManager.getOne.mockReturnValueOnce({
			...user,
			verificationCode: encryption.encryptData({ text: '345287' })
		});
		const spy = jest.spyOn(encryption, 'decryptData');
		const data = { id: 'x7i9-3l-n3k4-3i8bi2', verificationCode: '345287' };
		await checkUserVerificationCode(data);
		expect(spy).toHaveBeenCalledWith(
			expect.objectContaining({ encryptedText: expect.any(String) })
		);
	});
});

describe('#verifyUserLoginCredentials', () => {
	const spy = jest.spyOn(bcryptjs, 'compareSync');

	test('should throw error if username, email and phoneNumber is null', (done) => {
		verifyUserLoginCredentials({
			password: 'jellybean',
			email: '',
			phoneNumber: '',
			username: ''
		}).catch((error) => {
			expect(error.message).toBe('phoneNumber, email or username is required');
			done();
		});
	});

	test('should throw if there is no password', (done) => {
		verifyUserLoginCredentials({
			password: '',
			email: 'test@test.com',
			phoneNumber: '',
			username: ''
		}).catch((error) => {
			expect(error.message).toBe('password is required');
			done();
		});
	});

	test('should throw error if user is not found', (done) => {
		(runQuery.mockResolvedValueOnce as any)(null);
		verifyUserLoginCredentials({
			password: 'testing',
			email: 'test@test.com',
			phoneNumber: '',
			username: 'tester'
		}).catch((error) => {
			expect(error.message).toBe('Invalid user account');
			done();
		});
	});

	test('should throw error if password does not match', () => {
		spy.mockReturnValueOnce(false);
		entityManager.getOne.mockReturnValueOnce(createUserFixture());
		const credentials = {
			password: 'testing',
			email: 'test@test.com',
			phoneNumber: '',
			username: 'tester'
		};
		expect(verifyUserLoginCredentials(credentials)).rejects.toThrowErrorMatchingSnapshot();
	});

	test('should return user details with token', async () => {
		spy.mockReturnValueOnce(true);
		const user = createUserFixture();
		entityManager.getOne.mockReturnValueOnce(user);
		const value = await verifyUserLoginCredentials({
			password: user.password,
			email: user.email,
			phoneNumber: '',
			username: user.username
		});
		expect(value).toEqual(expect.objectContaining({ token: expect.any(String) }));
	});
});

describe('#getUserAccountWithCredentials', () => {
	const userProfile = createUserFixture();
	const credentials = {
		username: userProfile.username,
		phoneNumber: userProfile.phoneNumber,
		email: userProfile.email
	};

	beforeAll(() => {
		const verificationCode = encryption.encryptData({ text: '123456' });
		Object.assign(userProfile, { verificationCode });
	});

	test('should throw if user is not found', () => {
		entityManager.getOne.mockReturnValueOnce(null);
		expect(getUserAccountWithCredentials(credentials)).rejects.toThrowErrorMatchingSnapshot();
	});

	test('should add token to the return object', async () => {
		entityManager.getOne.mockReturnValueOnce(userProfile);
		const user = await getUserAccountWithCredentials(credentials);
		expect(user.token).toEqual(expect.any(String));
	});

	test('should not return the password', async () => {
		entityManager.getOne.mockReturnValueOnce(userProfile);
		const user = await getUserAccountWithCredentials(credentials);
		expect(user).not.toHaveProperty('password');
	});

	test('should update user with new verification code', async () => {
		entityManager.getOne.mockReturnValueOnce(userProfile);
		await getUserAccountWithCredentials(credentials);
		expect(runQuery).toHaveBeenNthCalledWith(
			2,
			expect.any(Function),
			expect.arrayContaining([
				expect.objectContaining({
					id: expect.any(String),
					verificationCode: expect.any(String)
				})
			])
		);
	});

	test('should create new verification code', async () => {
		entityManager.getOne.mockReturnValueOnce(userProfile);
		const spy = jest.spyOn(utils, 'generateCodeFromNumber');
		await getUserAccountWithCredentials(credentials);
		expect(spy).toHaveReturnedWith(expect.any(String));
	});

	test('should encrypt the verification code', async () => {
		entityManager.getOne.mockReturnValueOnce(userProfile);
		const spy = jest.spyOn(utils, 'generateCodeFromNumber');
		spy.mockReturnValueOnce('678902');
		await getUserAccountWithCredentials(credentials);
		expect(runQuery).toHaveBeenNthCalledWith(
			2,
			expect.any(Function),
			expect.arrayContaining([
				{ id: userProfile.id, verificationCode: expect.not.stringMatching('678902') }
			])
		);
	});
});
