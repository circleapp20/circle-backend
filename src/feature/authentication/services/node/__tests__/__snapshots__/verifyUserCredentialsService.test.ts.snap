// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#checkUserVerificationCode should throw if id is empty 1`] = `"user id and verification code are required"`;

exports[`#checkUserVerificationCode should throw if user is null 1`] = `"Invalid user account"`;

exports[`#checkUserVerificationCode should throw if verificationCode is empty 1`] = `"user id and verification code are required"`;

exports[`#getUserAccountWithCredentials should throw if user is not found 1`] = `"Invalid user account"`;

exports[`#verifyUserLoginCredentials should throw error if password does not match 1`] = `"Invalid user account"`;
